### PR TITLE
Initial support for deploying MEANJS to Cloud Foundry

### PR DESCRIPTION
**Pull request by ryanjbaxter**
Initial support for deploying MEANJS for Cloud Foundry.  Additional documentation can be found in #716.

### CODE REVIEW

**Comment by lirantal** [comment]
sure, although we discussed in https://github.com/meanjs/mean/pull/716 that @codydaig needs some time to review this change so we'll probably need to merge it after the 0.4.0 release

File: README.md [hunk 1]
```diff
@@ -171,6 +171,39 @@ In the docs we'll try to explain both general concepts of MEAN components and gi
...
+* Install the [Cloud Foundry command line tools](http://docs.cloudfoundry.org/devguide/installcf/install-go-cli.html).
+* Now you need to log into Cloud Foundry from the Cloud Foundry command line.
+  *  If you are using Pivotal Web Services run `$ cf login -a api.run.pivotal.io`.
+  *  If you are using IBM Bluemix run `$ cf login -a api.ng.bluemix.net`.
+* Create a Mongo DB service, IBM Bluemix and Pivotal Web Services offer a free MongoLabs service.
+  *  `$ cf create-service mongolab sandbox mean-mongo`
+* Clone the GitHub repo for MEANJS if you have not already done so
+  * `$ git clone https://github.com/meanjs/mean.git && cd mean`
+* Run the Grunt Build task to build the optimized JavaScript and CSS files
+  * `$ grunt build`
```
Comment (codydaig) [inline README.md]: @ryanjbaxter In this pull request you say you have to run grunt build, but in the pull request to the documentation you don't mention anything about grunt build. Is it required?

File: README.md [hunk 2]
```diff
@@ -171,6 +171,39 @@ In the docs we'll try to explain both general concepts of MEAN components and gi
...
+
+* Install the [Cloud Foundry command line tools](http://docs.cloudfoundry.org/devguide/installcf/install-go-cli.html).
+* Now you need to log into Cloud Foundry from the Cloud Foundry command line.
+  *  If you are using Pivotal Web Services run `$ cf login -a api.run.pivotal.io`.
+  *  If you are using IBM Bluemix run `$ cf login -a api.ng.bluemix.net`.
+* Create a Mongo DB service, IBM Bluemix and Pivotal Web Services offer a free MongoLabs service.
+  *  `$ cf create-service mongolab sandbox mean-mongo`
+* Clone the GitHub repo for MEANJS if you have not already done so
+  * `$ git clone https://github.com/meanjs/mean.git && cd mean`
+* Run the Grunt Build task to build the optimized JavaScript and CSS files
```
Comment (codydaig) [inline README.md]: @ryanjbaxter If we do need to run grunt build, then you should include a step for npm install.

File: .cfignore [hunk 3]
```diff
@@ -0,0 +1,10 @@
++# List of files and directories to ignore when deploying to Cloud Foundry
++.DS_Store
++.nodemonignore
++.sass-cache/
++npm-debug.log
++node_modules/
```
Comment (codydaig) [inline .cfignore]: When I perform a `cf push` the node_modules folder gets uploaded as well.

**Comment by codydaig** [comment]
@lirantal - Is the addition to the README okay, or do you only want it in the docs on the website?
IMO, I'd like to see a "Deployment" page on the website for these types of tutorials and just put a link in the README to the deployment page on the website. 

@ryanjbaxter Once we hear back about the README, just squash the commits and this will be ready to merge.

**Comment by lirantal** [comment]
@codydaig I definitely get your point but we already have today a few other 'deployment places' paths, right?

**Comment by codydaig** [comment]
@lirantal Currently the other other deployment documentation is with Docker, even though we also have support build in for Heroku. I'm okay with the addition to the README as well as the additional info on the website for now and we can discuss changes further down the road as more deployment options become documented. https://github.com/meanjs/mean#development-and-deployment-with-docker

**Comment by lirantal** [comment]
@codydaig I think it's ok to have it in the README for now
@ryanjbaxter don't forget to squash your commits

**Comment by ryanjbaxter** [comment]
@lirantal if everyone is fine with this and there are no other changes I will attempt to squash the commits, I just dont want to squash them and then do it again

**Comment by ryanjbaxter** [comment]
@lirantal commits squashed

**Comment by ryanjbaxter** [comment]
Actually hold up on the commit, I want to make one small change to the manifest file

**Comment by ryanjbaxter** [comment]
OK small change made, we are good to go

**Comment by codydaig** [comment]
@ryanjbaxter What change did you make?

**Comment by ryanjbaxter** [comment]
@codydaig I added some more information to the manifest file.  This should help simplify deployments in the future.  We have a nice feature in Bluemix where you can have a button that will do the deployment for you when clicked.  The manifest changes support that functionality.

**Comment by ryanjbaxter** [comment]
When can we get these changes merged in?  I see that 0.4.0 was released.

**Comment by codydaig** [comment]
@ryanjbaxter I need to test this again with the latest changes that you made.

**Comment by lirantal** [comment]
@codydaig let us know please if we can merge this change in and maybe we can figure out a quick way to merge it into master without requiring a re-submit of the PR

**Comment by codydaig** [comment]
@lirantal LGTM

**Comment by lirantal** [comment]
@ryanjbaxter I merged it to master for you: https://github.com/meanjs/mean/commit/efd3ee4ce842861bc1044f0c8b5883bb715d531f

@codydaig what about https://github.com/meanjs/mean/pull/716 documentation? can we merge that too or does it need more work?

**Comment by codydaig** [comment]
@lirantal This PR can be closed then. 

@ryanjbaxter Let's take the documentation conversation to #716. Have some final questions/review before it's ready to be merged.

**Pull request closed without merging**